src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -665,10 +665,16 @@ void G1Policy::record_pause_start_time() {
   Ticks now = Ticks::now();
   _cur_pause_start_sec = now.seconds();
 
-  double prev_gc_cpu_pause_end_ms = _analytics->gc_cpu_time_at_pause_end_ms();
-  double cur_gc_cpu_time_ms = _analytics->gc_cpu_time_ms();
-
-  double concurrent_gc_cpu_time_ms = cur_gc_cpu_time_ms - prev_gc_cpu_pause_end_ms;
+  // Do not read CPU times for concurrent GC threads if the VM is shutting down,
+  // as these threads may have already terminated, resulting in crashes.
+  // Only compute concurrent GC CPU time if the VM is not in the process of shutting down.
+  double concurrent_gc_cpu_time_ms = 0.0;
+  if (!_g1h->is_shutting_down()) {
```

Comment:
Why is `is_shutting_down` checked here? Can its caller ensure we are not in shutdown, e.g. `G1CollectedHeap::do_collection_pause_at_safepoint`?

The current method name kind of suggests that we are in normal gc-pause, not shutdown process, IMO.

Reply:
Is it possible for following a situation to occur?
1. thread A run to here
2. thread B call G1CollectedHeap::stop()
3. then after a while, thread A execute following 2 lines.

If it's possible, could it still cause the similar crash in some extreme situation?